extensions/ui-extension/src/targets/purchase/checkout/shipping-option-list/useVatApi.tsx

```
@@ -0,0 +1,92 @@
+import { APP_CODE } from '@/helpers/shop';
+import { useIfixitBaseUrl } from '@/hooks/useIfixitBaseUrl';
+import { useSessionToken } from '@shopify/ui-extensions-react/checkout';
+import { z } from 'zod';
+
+type ValidationApiInput = {
+   company: string;
+   country_code: string;
+   vat_id: string;
+};
+const VatValidationSuccessSchema = z.object({
+   errors: z.undefined(),
+   valid: z.boolean(),
+   validFormat: z.boolean(),
+   vatId: z.string(),
+});
+const VatValidationErrorCodeSchema = z.enum([
+   'INVALID_COMPANY',
+   'INVALID_COUNTRY',
+   'SERVICE_UNAVAILABLE',
+]);
+type VatValidationErrorCode = z.infer<typeof VatValidationErrorCodeSchema>;
+// Keep in sync with PHP `IfixitVatValidationErrorCode` enum
+const VatValidationErrorSchema = z.object({
+   errors: z.array(VatValidationErrorCodeSchema),
+});
+const VatValidationResultSchema = z.union([
+   VatValidationSuccessSchema,
+   VatValidationErrorSchema,
+]);
+type VatValidationResult = z.infer<typeof VatValidationResultSchema>;
+
+export function useVatApi() {
+   const ifixitBaseUrl = useIfixitBaseUrl();
+   const sessionToken = useSessionToken();
+   return { revokeApi, validateApi };
+
+   async function revokeApi() {
+      const token = await sessionToken.get();
+      const res = await fetch(
+         `${ifixitBaseUrl}/api/2.0/store/checkout/vat/validate/${APP_CODE}`,
+         {
+            headers: {
+               Authorization: `Bearer ${token}`,
+            },
+            method: 'DELETE',
+         }
+      );
+      let json: { message?: string } = {};
+      try {
+         json = (await res.json()) as { message?: string };
+      } catch (e) {}
```

should we also throw an error if this fails?

---

Yes for the validation endpoint, which always returns json, I made that change thank you. The delete endpoint only returns json on error, so we can ignore a non-json response on success.